app/controllers/application_controller.rb

```diff
@@ -37,10 +38,14 @@ def context
   end
   helper_method :context
 
+  def shown?
+    NavbarHelper.show_navbar?(params[:controller], params[:action], context.system_settings.display_navbar?)
+  end
+
```

Comment:
I like this direction because it means `show_navbar?` doesn't need to be mixed into every controller.

Pushing that logic a bit further, i don't think we really need the `shown?` method either.

We could instead move this logic directly into `layouts/application.html.erb` since that's the only place it is used:
``` erb
  <%# layouts/application.html.erb %>
  <% if shown_navbar? controller_name, action_name, context.system_settings.display_navbar? %>
    <%= render "layouts/navbar" %>
  <% end %> 
```
This way, nothing extra is added to `ApplicationController`.

Note also the use of `controller_name` and `action_name` above.

Reply:
Done :+1: